src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -106,13 +106,13 @@ class SessionService(
     }.flatten()
 
     val sessionSlots = getSessionSlots(visitSessions)
-    val nonAssociationConflictSessions = getNonAssociationSessions(visitSessions, sessionSlots, prisonerId)
+    val nonAssociationConflictSessions = getNonAssociationSessions(visitSessions, prisonerId)
     val doubleBookings = getDoubleBookingSessions(visitSessions, sessionSlots, prisonerId)
 
     return visitSessions.filterNot {
-      filterOutNonAssociations(nonAssociationConflictSessions, it)
+      hasNonAssociationConflict(nonAssociationConflictSessions, it) && policyFilterNonAssociation
```

Comment:
while we working in this area, could we clean up the return to do the logic before hand? Theres a lot going on during the result, could we move it upwards and rename the `it` to `visitSessions` ? 

Make it easier to follow and read in the future?

Reply:
Yeha I agree code needs refactoring, we have been doing this on and off since we joined but we prefer to do this in it's own ticket.